### ISSUE TITLE
[BUG] Latest release "auto" GenomicArray crashes existing code due to finite length chromosome if 1st accessed via setter

### ISSUE THREAD

**Issue by davmlaw** [htseq/htseq#38]
The following code ran without crashing on all previous versions of HTSeq (ie up to and including Ie up to and including v0.13.5 (Dec 29, 2020)

```
import HTSeq
from importlib import metadata                                                                                           

print(f'HTSeq version: {metadata.version("HTSeq")}')        


ga = HTSeq.GenomicArray("auto")                                                                                          

iv = HTSeq.GenomicInterval("1", 100, 150, "+")                                                               
iv2 = HTSeq.GenomicInterval("1", 200, 300, "+")                                                                          

ga[iv] = 2
data = ga[iv2]

print("Ran ok!")
```

but now crashes:

```
HTSeq version: 1.99.2
Traceback (most recent call last):
  File "./htseq_test.py", line 15, in <module>
    data = ga[iv2]
  File "src/HTSeq/_HTSeq.pyx", line 701, in HTSeq._HTSeq.GenomicArray.__getitem__
  File "src/HTSeq/_HTSeq.pyx", line 480, in HTSeq._HTSeq.ChromVector.__getitem__
IndexError: stop too large
```

This was due to commit:

https://github.com/htseq/htseq/commit/b2d20a3d50a23e0e5f2634b960dec480f863662e#diff-0f12eb8d539626111020f6eaf57e93427ae1fb352b95688ba0e8e0c9b09012c1L639

Which changed the GenomicArray setter to make a non-infinite length new chromosome.

GenomicArray used to create infinite length chromosomes in this case, and still does in all other instances, eg explicitly naming chroms makes them infinite length:

```
In [1]: import HTSeq                                                                                                                     

In [2]: HTSeq.__version__                                                                                                                
Out[2]: '1.99.2'

In [3]: ga_chroms = HTSeq.GenomicArray(["1"])                                                                                            

In [4]: ga_chroms["1"]                                                                                                                   
Out[4]: 
{'+': <ChromVector object, 1:[0,Inf)/+, step>,
 '-': <ChromVector object, 1:[0,Inf)/-, step>}
```

Accessing the unknown chromosome for the first time via a get makes it infinite:

```
In [5]: iv = HTSeq.GenomicInterval("1", 1, 100, "+")                                                                                     

In [6]: ga_getter = HTSeq.GenomicArray("auto")                                                                                           

In [7: ga_getter[iv]                                                                                                                   
Out[7]: <ChromVector object, 1:[1,100)/+, step>

In [8]: ga_getter["1"]                                                                                                                  
Out[8]: 
{'+': <ChromVector object, 1:[0,Inf)/+, step>,
 '-': <ChromVector object, 1:[0,Inf)/-, step>}
```

However if you access it for the first time via a set, it is only of that interval's size:

```
In [9]: ga_setter = HTSeq.GenomicArray("auto")                                                                                           

In [10]: ga_setter[iv] = 42                                                                                                               

In [11]: ga_setter["1"]                                                                                                                   
Out[11]: 
{'+': <ChromVector object, 1:[1,100)/+, step>,
 '-': <ChromVector object, 1:[1,100)/-, step>}
```

**Expected Result**

Old code written for HTSeq continues to run without crashing.

This means Genomic Array "auto" chromosomes need to be consistently infinite size when created via SET consistent with other methods

**Software versions**
- `HTSeq` - 1.99.2
- `Python` - Python 3.8.10
- operating system - Ubuntu 20.04.3

**Comment by kpillman** [comment]
Side observation: The behaviour described above does not occur for version 0.13.0 (in which, for a "auto"-type GenomicArray, adding a GenomicInterval on a new chromosome created the chromosome vector of start=0, end=Inf). 

Thanks for your help!

**Comment by iosonofabio** [comment]
Thank you! It sounds like you could make a PR for this? That would speed up adoption...

**Comment by davmlaw** [comment]
I renamed "regions_array" to "ga_setter" in comment above. If you access ga_setter like that, it fails, while the other genomic arrays work.

I'll submit a pull request shortly.

Basically the problem is a lack of symmetry, if you call ```__getitem__```  it creates a new chromosome via  ```self.add_chrom(index.chrom)``` while the logic for ```__setitem__``` is much more complicated. Calling a getter before a setter shouldn't change the logic of how genomic arrays return data

Another example is a lack of symmetry in provided chromosomes vs auto:

```
In [19]: ga = HTSeq.GenomicArray(["1"])                                                                                                                               

In [20]: ga[iv] = 1                                                                                                                                                   

In [21]: list(ga.steps())                                                                                                                                             
Out[21]: 
[(<GenomicInterval object '1', [0,1), strand '+'>, 0.0),
 (<GenomicInterval object '1', [1,100), strand '+'>, 1.0),
 (<GenomicInterval object '1', [100,Inf), strand '+'>, 0.0),
 (<GenomicInterval object '1', [0,Inf), strand '-'>, 0.0)]

In [22]: ga_auto = HTSeq.GenomicArray("auto")                                                                                                                         

In [23]: ga_auto[iv] = 1                                                                                                                                              

In [24]: list(ga_auto.steps())                                                                                                                                        
Out[24]: 
[(<GenomicInterval object '1', [1,100), strand '+'>, 1.0),
 (<GenomicInterval object '1', [1,100), strand '-'>, 0.0)]
```

I think these should act the same.

My fix is to make the add_chrom calls from get/set be the same:

https://github.com/davmlaw/htseq/commit/55d841b6b621c881e6e3b146dd5eabc146fb5a91#diff-0f12eb8d539626111020f6eaf57e93427ae1fb352b95688ba0e8e0c9b09012c1

**Comment by iosonofabio** [comment]
> Calling a getter before a setter shouldn't change the logic of how genomic arrays return data

fair enough

Thank you for reporting, looking forward to seeing the PR. I'll push a few tests to it and if it passes CI I'll merge it and will make it into the next release (expected in a month or so). I'll leave this issue open until then.

**Comment by iosonofabio** [comment]
Fixed via #39 

Thanks again for reporting.

**Comment by davmlaw** [comment]
Thanks!
